cmd/ingester/app/consumer/consumer.go

```diff
@@ -110,8 +114,16 @@ func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 
 	deadlockDetector := c.deadlockDetector.startMonitoringForPartition(pc.Partition())
 	defer deadlockDetector.close()
-
+	var rateLimiter *rateLimiter
+	if c.maxReadsPerSecond > 0 {
```

Comment:
Why allow the `rateLimiter` to be `nil`?
If we consolidate the logic for when `maxReadsPerSecond <= 0` into the rateLImiter, then we can remove the `if` in L124

Reply:
I see - IMHO a cleaner way of solving this would be to use a no-op rateLimiter implementation that is picked when the user doesn't require rate limiting

Reply:
Good idea!